applog/memory.go

```diff
@@ -118,7 +119,8 @@ func (s *memoryLogService) List(args appTypes.ListLogArgs) ([]appTypes.Applog, e
 	var count int
 	for current := buffer.end; count < args.Limit; {
 		if (args.Source == "" || (args.Source == current.log.Source) != args.InvertFilters) &&
-			(args.Unit == "" || (args.Unit == current.log.Unit) != args.InvertFilters) {
+			(args.Unit == "" || (args.Unit == current.log.Unit) != args.InvertFilters) &&
+			(args.Level == 0 || (args.Level == current.log.Level) != args.InvertFilters) {
```

Comment:
We need some tests for this check and also for the equivalent change on the mongodb implementation. Adding a new method to the `ServiceSuite` in `applog/service_test.go` will automatically run the test on both implementations.